File: haskell/app/Menus/Inicio.hs
-- Menu que inicializa o sistema
module Menus.Inicio where

-- Importações necessárias para o funcionamento adequado
import System.Exit (exitSuccess)
import Menus.Util (escolherOpcoes)
import Menus.Cadastro (menuCadastro)
import Menus.Login (menuLogin)
import qualified Menus.Cores as Cores
import System.Console.ANSI (clearScreen)

-- Tela inicial, menu que mostra as funcionalidades para o usuário
menu :: [String]
menu = [
        "╔═══════════════════════════════════════════════════════════════════════════════╗",
        "║                                                                               ║",
        "║                 ______  __                 __               _______           ║",
        "║               .' ___  |[  |               [  |  _          |_   __ |          ║",
        "║             / .'   |_| | |  .--.    .--.  | | / ] __   _    | |__) |          ║",
        "║             | |        | |/ .'`| |/ .'`| || '' < [  | | |   |  ___/           ║",
        "║             | `.___.'| | || |__. || |__. || |`| | | |_/ |, _| |_              ║",
        "║              `.____ .'[___]'.__.'  '.__.'[__|  |_]'.__.'_/|_____|             ║",
        "║                                                                               ║",
        "║       Menu:                                                                   ║",
        "║          " ++ Cores.reseta ++ Cores.negrito ++ "1. Login      " ++ Cores.laranja ++ "                                                       ║",
        "║          " ++ Cores.reseta ++ Cores.negrito ++ "2. Cadastro   " ++ Cores.laranja ++ "                                                       ║",
        "║          " ++ Cores.reseta ++ Cores.negrito ++ "3. Sair       " ++ Cores.laranja ++ "                                                       ║",
        "║                                                                               ║",
        "║                                                                               ║",
        "║       Digite a opção:                                                         ║",
        "╚═══════════════════════════════════════════════════════════════════════════════╝"
    ]

-- Carrega o menu inicial para que o usuário consiga acessar uma determinada funcionalidade 
menuInicio :: IO()
menuInicio = do
    clearScreen
    putStrLn $ Cores.laranja ++ unlines menu ++ Cores.reseta
    escolherOpcoes [
        menuLogin,
        menuCadastro,
        sair
     ]
    -- loop infinito, já que o unico jeito de sair é utilizando a opção de sair
    menuInicio

-- Função que interrompe o loop inicializado e finaliza a execução do menu principal
sair :: IO()
sair = do
    putStrLn "Até Mais!"
    exitSuccess

File: prolog/Menus/Inicio.pl
:- module(inicio, [menuInicio/0]).

:- use_module('Menus/Utils.pl').
:- use_module('Menus/Login.pl', [menuLogin/0]).
:- use_module('Menus/Cadastro.pl', [menuCadastro/0]).

printBanner :- 
    print_cor("&l╔═══════════════════════════════════════════════════════════════════════════════╗\n"),
    print_cor("║                                                                               ║\n"),
    print_cor("║                 ______  __                 __               _______           ║\n"),
    print_cor("║               .' ___  |[  |               [  |  _          |_   __ |          ║\n"),
    print_cor("║             / .'   |_| | |  .--.    .--.  | | / ] __   _    | |__) |          ║\n"),
    print_cor("║             | |        | |/ .'`| |/ .'`| || '' < [  | | |   |  ___/           ║\n"),
    print_cor("║             | `.___.'| | || |__. || |__. || |`| | | |_/ |, _| |_              ║\n"),
    print_cor("║              `.____ .'[___]'.__.'  '.__.'[__|  |_]'.__.'_/|_____|             ║\n"),
    print_cor("║                                                                               ║\n"),
    print_cor("║       Menu:                                                                   ║\n"),
    print_cor("║          &r&b1. Login    &r&l                                                         ║\n"),
    print_cor("║          &r&b2. Cadastro &r&l                                                         ║\n"),
    print_cor("║          &r&b3. Sair     &r&l                                                         ║\n"),
    print_cor("║                                                                               ║\n"),
    print_cor("║                                                                               ║\n"),
    print_cor("║       Digite a opção:                                                         ║\n"),
    print_cor("╚═══════════════════════════════════════════════════════════════════════════════╝&r\n").

menuInicio :- 
    clear_screen,
    printBanner,
    read_number(I), 
    (I == 1 -> menuLogin; I == 2 -> menuCadastro; I == 3 -> sair; writeln("Erro: Opção invalida!")),
    menuInicio.

sair:- writeln("Até mais!"), halt(0).